src/winerror.cr

```diff
@@ -0,0 +1,2104 @@
+class WinError < Exception
```

Comment:
I thought about this. The problem is that errno also exists on Windows and it's used for C functions like _open  [link](https://msdn.microsoft.com/en-us/library/5814770t.aspx) and Errno class is already working for Windows. So mostly Win32 API functions use GetLastError, "posix" subsystem (some functionality will be used in crystal) uses errno. So how to use one class for these different things simultaneously?